src/Plugin/AddPathPlugin.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Add base path to the request URI. Useful for base API URLs like http://domain.com/api.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class AddPathPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $host;
+
+    /**
+     * @param UriInterface $host
+     */
+    public function __construct(UriInterface $host)
```

Comment:
Well, actually we don't need a host here at all. The user just has to pass a path (not a host) and I am not sure that actually needs any validation at all.